src/analyses/base.ml

```diff
@@ -2501,9 +2501,27 @@ struct
           | _ ->
             set ~man st lv_a lv_typ (VD.top_value (unrollType lv_typ))
         end
-    in 
+    in
+    let alloc ?(offset=`NoOffset) loc size =
+      let bytes = eval_int ~man st size in
+      let is_zero = ID.equal_to Z.zero bytes in
+      let heap_var =
+        let include_null = get_bool "sem.malloc.fail" || (is_zero <> `Neq && get_string "sem.malloc.zero" <> "pointer") in
+        let include_pointer = (is_zero <> `Eq || get_string "sem.malloc.zero" <> "null") in
```

Comment:
All these negations make this very difficult to follow. Especially because both things have values where some are included in another.
Maybe less negations or pattern matching would make this easier to understand.

Reply:
I tried to make more explicit what happens.